base/complex.jl

```diff
@@ -259,7 +259,11 @@ julia> conj(1 + 3im)
 conj(z::Complex) = Complex(real(z),-imag(z))
 abs(z::Complex)  = hypot(real(z), imag(z))
 abs2(z::Complex) = real(z)*real(z) + imag(z)*imag(z)
-inv(z::Complex)  = conj(z)/abs2(z)
+function inv(z::Complex)
+    a, b = reim(z)
+    (isinf(a) || isinf(b)) && return complex(copysign(zero(a), a), flipsign(-zero(b), b))
```

Comment:
`|` rather than `||` since we don't want short-circuit evaluation here (in the common case both will be `false`), I think?

Reply:
Also, I think the imaginary part should be:
```jl
flipsign(-copysign(zero(b), b), a)
```

Reply:
In `(a + bi) / (c + di)`, the imaginary part is `(bc - ad) / (c^2 + d^2)`, so with `a = 1` and `b = 0` the numerator is simply `-d`, isn't it?  The numerator of the real part is `cd`, so probably that sign above is wrong?  `copysign(zero(a), sign(a) * sign(b)` instead?

Reply:
Whoops, right.

Reply:
Ok, I changed `||` to `|` and left the rest as it was (apart from renaming `a` and `b` to `c` and `d` respectively), thanks!